StardewArchipelago/GameModifications/Modded/JunimoShopStockModifier.cs

```diff
@@ -71,17 +72,17 @@ private void GenerateBlueJunimoStock(ShopData shopData)
             var fishData = DataLoader.Fish(Game1.content);
             shopData.Items.Clear();
 
-            foreach (var fish in fishData)
+            foreach (var (id, fishInfo) in fishData)
             {
                 string[] fishSeasons = null;
-                if (fishData[fish.Key].Split("/")[1] != "trap")
+                if (fishData[id].Split("/")[1] != "trap")
                 {
-                    fishSeasons = fishData[fish.Key].Split("/")[6].Split(" ");
+                    fishSeasons = fishData[id].Split("/")[6].Split(" ");
                 }
-                var fishItem = _stardewItemManager.GetObjectById(fish.Key);
-                var completeCondition = fishSeasons is not null ? 
-                $"{GameStateConditionProvider.CreateSeasonsCondition(fishSeasons)}, RANDOM 0.4 @addDailyLuck, PLAYER_HAS_CAUGHT_FISH Current {fish.Key}" : 
-                "RANDOM 0.4 @addDailyLuck, PLAYER_HAS_CAUGHT_FISH Current {fish.Key}";
+                var fishItem = _stardewItemManager.GetObjectById(id);
+                var conditionIfNotNull = $"{GameStateConditionProvider.CreateSeasonsCondition(fishSeasons)}, RANDOM 0.4 @addDailyLuck, PLAYER_HAS_CAUGHT_FISH Current {id}";
```

Comment:
You split the ternary into its parts, which is good, but it's not what I asked for lol. You still have the section `RANDOM 0.4 @addDailyLuck, PLAYER_HAS_CAUGHT_FISH Current {id}` being defined twice, and therefore prone to errors

Reply:
Now that I think about it, splitting this ternary isn't good, because you will now encounter NullRef exceptions, as both sides are always evaluated.
So it should probably be put back together, and just the change I had asked for should be done